exporter/tanzuobservabilityexporter/metrics_exporter.go

```diff
@@ -58,20 +57,17 @@ type metricsConsumerCreator func(hostName string, port int, settings component.T
 	*metricsConsumer, error)
 
 func newMetricsExporter(settings component.ExporterCreateSettings, c config.Exporter, creator metricsConsumerCreator) (*metricsExporter, error) {
-	cfg, ok := c.(*Config)
+	cfgPtr, ok := c.(*Config)
 	if !ok {
 		return nil, fmt.Errorf("invalid config: %#v", c)
 	}
-	endpoint, err := url.Parse(cfg.Metrics.Endpoint)
+	_, _, metricsURL, err := cfgPtr.withoutMissingFields()
```

Comment:
See my comment in `exporter.go` regarding using the config's struct values directly.